concent_api/core/utils.py

```diff
@@ -140,3 +143,15 @@ def is_given_golem_messages_version_supported_by_concent(
         if not is_protocol_version_compatible(golem_message_version):
             return False
         return True
+
+
+def generate_uuid(last_char: Optional[str] = None) -> str:
+    random.seed(0)
+    random_bits = "%32x" % random.getrandbits(128)
```

Comment:
I you initialize the generator every time, you'll get the same random number every time:

``` python
>>> import random
>>> random.seed(0)
>>> random.getrandbits(128)
302934307671667531413257853548643485645
>>> random.seed(0)
>>> random.getrandbits(128)
302934307671667531413257853548643485645
>>> random.seed(0)
>>> random.getrandbits(128)
302934307671667531413257853548643485645
>>> random.seed(0)
>>> random.getrandbits(128)
302934307671667531413257853548643485645
>>> random.seed(0)
>>> random.getrandbits(128)
302934307671667531413257853548643485645
>>> random.seed(0)
>>> random.getrandbits(128)
302934307671667531413257853548643485645
```

We want this instead:
``` python
>>> import random
>>> random.seed(0)
>>> random.getrandbits(128)
302934307671667531413257853548643485645
>>> random.getrandbits(128)
328530677494498397859470651507255972949
>>> random.getrandbits(128)
312458951358162775936600420074706364002
>>> random.getrandbits(128)
329325272347819381495408738591077500473
>>> random.getrandbits(128)
303416399459424055589361154128267592196
```

You should initialize it globally, but in such a way that it's only done when tests run (we don't want random numbers to be predictable when we're not running tests). @dybi maybe you know a good place for global test initialization?

Reply:
Wow, right, good point.
What do you think if instead of function we create class as singleton with one method to generate random uuid. This way there is no need to initialize generator with other place, everything will be nice and in one class. And this singleton only once, when created will call `random.seed(0)` and then only `random.getrandbits(128)`. What do you think?

Reply:
let's change this functions signature and add an optional parameter `seed = None`. If 
seed is `None`:
```python
if seed is None:
     seed = get_current_utc_timestamp()
random.seed(seed)
```

Also this function was intended to be a helper test function. Right now, it has grown too big to be a single function. So, core functionality with generating uuid based on seed should stay here in `core.utils`, but the part with substituting last character should be moved to another function, used in tests only. The latter could be `generate_uuid_for_tests` or `create_uuid_for tests` or sth similar and it should be placed in `core.tests.utils`

Reply:
done